src/plugins/intel_gpu/src/graph/include/primitive_inst.h

```diff
@@ -212,6 +212,10 @@ class primitive_inst {
     void build_deps();
     void do_runtime_skip_reorder();
     void do_runtime_in_place_concat();
+    void set_output_buffer_used_for_next_input(const primitive_id& input_prim) {
+        _impl_params->output_buffer_used_for_next_input = input_prim;
+    }
```

Comment:
How about having some mapping at the network level to configure such dependencies, instead of using kernel_impl_params? Wouldn't that be more straightforward? And also this approach would allow to reset this map every iteration and avoid output's buffer reallocation in case of changes in output's reuse policy (if there will be any...)

Reply:
I also thought about it, but I needed to pass something to the allocate_output of network pointer or some other flag because it is a static function. So I did this for less change. How do you think?